x-pack/plugins/fleet/server/services/agent_policies/package_policies_to_agent_inputs.ts

```diff
@@ -104,8 +101,8 @@ export const storedPackagePolicyToAgentInputs = (
 };
 
 export const storedPackagePoliciesToAgentInputs = async (
-  soClient: SavedObjectsClientContract,
   packagePolicies: PackagePolicy[],
+  packageInfoCache: Map<string, PackageInfo>,
```

Comment:
These policy compilation helpers now make use a pre-fetched cache of `PackageInfo` objects for each package on a given policy, rather than fetching every single package on-demand.